EventsExpress.Core/Services/LocationService.cs

```diff
@@ -41,23 +41,35 @@ public async Task<Guid> Create(LocationDto locationDTO)
             return result.Id;
         }
 
-        public LocationDto LocationByPoint(Point point)
+        public Guid? LocationByPoint(Point point)
         {
-            var locationDTO = Mapper.Map<LocationDto>(Context.EventLocations
+            var locationDto = Mapper.Map<LocationDto>(Context.Locations
                 .Where(e =>
                     e.Point.X == point.X &&
                     e.Point.Y == point.Y)
                 .FirstOrDefault());
-            return locationDTO;
+            return locationDto.Id;
         }
 
-        public LocationDto LocationByURI(string uri)
+        public Guid? LocationByURI(string uri)
         {
-            var locationDTO = Mapper.Map<LocationDto>(Context.EventLocations
+            var locationDTO = Mapper.Map<LocationDto>(Context.Locations
                 .Where(e =>
                    e.OnlineMeeting == uri)
                 .FirstOrDefault());
-            return locationDTO;
+            return locationDTO.Id;
+        }
+
+        public async Task<Guid> AddLocationToUser(LocationDto locationDto)
+        {
+            if (locationDto.Type == LocationType.Map)
+            {
+                return LocationByPoint(locationDto.Point) ?? await Create(new LocationDto { Id = locationDto.Id, Point = locationDto.Point, OnlineMeeting = null, Type = locationDto.Type });
+            }
+            else
+            {
+                throw new EventsExpressException("Location error");
```

Comment:
Change message to something like "Uri-based location is not applicable to users"